docs/error-codes.md

```diff
@@ -1827,6 +1827,37 @@ void TestMethod()
   }
   ```
 
+#### `IL2117`: Trim analysis: Methods 'method1' and 'method2' are both associated with state machine type 'type'. This is currently unsupported and may lead to incorrectly reported warnings.
+
```

Comment:
I don't expect to hit it, I just made the unexpected case more like the warning we would give if we saw multiple methods associated with the same state machine type, instead of throwing an exception. I'm personally fine with throwing an exception instead - should we take the same approach for IL2107 or is there a reason to take a different approach?

Reply:
I think your other comment https://github.com/dotnet/linker/pull/2722#discussion_r842688650 answers it - I'll file an issue.